backend/models/notification.js

```diff
@@ -81,81 +82,181 @@ module.exports = {
+		const data = { issuesId: [issueId] };
+		return this.upsertNotification(username, data, types.ISSUE_ASSIGNED, criteria);
 	},
 
-	insertModelUpdatedNotification: function(username, teamSpace, modelId, revision) {
-		const data = {teamSpace,  modelId, revision};
+	insertModelUpdatedNotification: function (username, teamSpace, modelId, revision) {
+		const data = { teamSpace, modelId, revision };
 		return this.insertNotification(username, types.MODEL_UPDATED, data);
 	},
 
-	removeIssueAssignedNotification:function(username, teamSpace, modelId, issueId) {
-		const criteria = {teamSpace,  modelId, issuesId:{$in: [issueId]}};
+	removeIssueNotification: function (username, teamSpace, modelId, issueId, issueType) {
+		const criteria = { teamSpace, modelId, issuesId: { $in: [issueId] } };
 
-		return getNotification(username, types.ISSUE_ASSIGNED, criteria).then(notifications => {
+		return getNotification(username, issueType, criteria).then(notifications => {
 			if (notifications.length === 0) {
 				return null;
 			} else {
 				const n = notifications[0];
 				const index = n.issuesId.findIndex(i => i === issueId);
 				n.issuesId.splice(index, 1);
-				const data = {issuesId : n.issuesId};
+				const data = { issuesId: n.issuesId };
 
 				if (data.issuesId.length === 0) {
 					return this.deleteNotification(username, n._id)
-						.then(() => ({deleted:true , notification: {_id: utils.objectIdToString(n._id) }}));
+						.then(() => ({ deleted: true, notification: { _id: utils.objectIdToString(n._id) } }));
 				}
 				return this.updateNotification(username, n._id, data).then(() => {
-					return {deleted:false , notification: utils.objectIdToString(n)};
+					return { deleted: false, notification: utils.objectIdToString(n) };
 				});
 			}
 		});
 	},
 
-	deleteNotification: function(username, _id) {
+	deleteNotification: function (username, _id) {
 		_id = utils.stringToUUID(_id);
 
 		return db.getCollection(NOTIFICATIONS_DB, username)
-			.then(c => c.deleteOne({_id}));
+			.then(c => c.deleteOne({ _id }));
 	},
 
-	deleteAllNotifications: function(username) {
+	deleteAllNotifications: function (username) {
 		return db.getCollection(NOTIFICATIONS_DB, username)
 			.then(c => c.deleteMany({}));
 	},
 
+	createUserNotification: async function (users, teamSpace, modelId, issueId) {
+		return Promise.all(users.map(u => {
+			return this.upsertIssueClosedNotification(u, teamSpace, modelId, issueId)
+				.then((n) => {
+					return ({ username: u, notification: n });
+				});
+		}));
+	},
+
+	/**
+	 * This function checks wether users are authorised to access a model and make changes.
+	 *
+	 * @param {Array} users The users who need to be notified.
+	 * @param {string} teamSpace The current Teamspace project.
+	 * @param {string} modelId The model of which to inform users of notification.
+	 * @returns {Array} Returns the users that pass the authorisation check.
+	 * */
+
+	allowAccess: async function (users, teamSpace, modelId) {
```

Comment:
Probably should be called `canUpdateModel`. It doesn't seem to be used by anything. Remove entirely? @sanmont3drepo